synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -349,6 +369,76 @@ Dock_PalEdit::add_color(const synfig::Color& x)
 	return size()-1;
 }
 
+bool
+Dock_PalEdit::check_hex_format(const gchar* hexcolor)
+{
+	//Check if clipboard is empty
+	if(hexcolor==NULL)
+		return false;
+
+	//Check if first character is a # for the correct format
+	if(hexcolor[0]!='#')
+		return false;
+
+	int strsize = strlen(hexcolor);
+
+	//Check size is correct for the format: #ffffff or #fff or #f
+	//these are the sizes supported by the Color::set_hex() function 
+	if(strsize!=7&&strsize!=4&&strsize!=2)
+		return false;
+
+	//Checking if all characters are between 0-F uppercase or lowercase in ascii
+	for(int i=1;i<strsize;i++){
+		char c = hexcolor[i];
+
+		//    0   -   9  || A   -   F  || a   -    f
+		if(!((47<c&&c<58)||(64<c&&c<71)||(96<c&&c<103)))
```

Comment:
You can use [std::isxdigit](https://en.cppreference.com/w/cpp/string/byte/isxdigit) here.

Reply:
the for loop has been updated to use that function. That's a really cool function I didn't know about 😅, thanks for bringing my attention to it.